packages/editor/src/components/parts/mail/MailAttachmentPart.tsx

```diff
@@ -29,70 +30,77 @@ const MailAttachmentsPart = () => {
 
 const MailAttachmentTable = () => {
   const { config, update } = useMailData();
+  type MailAttachment = { attachment: string };
+  const [data, setData] = useState<MailAttachment[]>(config.attachments.map(filename => ({ attachment: filename })));
 
-  const columns = useMemo<ColumnDef<string>[]>(
+  const columns = useMemo<ColumnDef<MailAttachment>[]>(
     () => [
       {
         id: 'attachment',
-        accessorFn: row => row,
+        accessorFn: row => row.attachment,
         header: () => <span>Attachments</span>,
-        cell: cell => <ScriptCell cell={cell} type='Attachment' browsers={['attr', 'func', 'type', 'cms']} />
+        cell: cell => (
+          <ScriptCell cell={cell} type='Attachment' browsers={['attr', 'func', 'type', 'cms']} placeholder={'Enter the Attachment'} />
+        )
       }
     ],
     []
   );
 
-  const addRow = () => {
-    update('attachments', [...config.attachments, '']);
+  const onChange = (change: MailAttachment[]) => {
+    setData(change);
+    const mappedData: string[] = change.map(attachment => attachment.attachment);
+    update('attachments', mappedData);
   };
 
-  const removeTableRow = (index: number) => {
-    const newData = [...config.attachments];
-    newData.splice(index, 1);
-    update('attachments', newData);
-  };
-
-  const table = useReactTable({
-    data: config.attachments,
+  const { table, rowSelection, setRowSelection, addRow, removeRow, showAddButton } = useResizableEditableTable({
+    data,
     columns,
-    getCoreRowModel: getCoreRowModel(),
-    meta: {
-      updateData: (rowId: string, _columnId: string, value: unknown) => {
-        const rowIndex = parseInt(rowId);
-        const newData = [...config.attachments];
-        newData.splice(rowIndex, 1, value as string);
-        update('attachments', newData);
-      }
-    }
+    onChange,
+    emptyDataObject: { attachment: '' }
   });
 
+  const tableActions =
+    table.getSelectedRowModel().rows.length > 0
+      ? [
+          {
+            label: 'Remove row',
+            icon: IvyIcons.Trash,
+            action: () => removeRow(table.getRowModel().rowsById[Object.keys(rowSelection)[0]].index)
+          }
```

Comment:
maybe instead of the removeRow function which is provided by the useResizableEditableTable we should just return this Remove ControlField object...

Reply:
Yes, however, I think I need to implement a function that returns this object instead of directly providing an object. This is because the removeRow function within the ControlField object requires an index that can only be passed from the component. So, if I provide a function where you can pass the index property and then return the corresponding ControlField, it should do the trick, correct?

Reply:
Nevermind, I made a thinking error. I can simply return the object without any issues 👍